### ISSUE TITLE
[source-quickbooks] not working after access token expires

### ISSUE THREAD

**Issue by hans-extenteam** [airbytehq/airbyte#45911]
### Connector Name

source-quickbooks

### Connector Version

3.0.3

### What step the error happened?

Updating the connector

### Relevant information

Added all OAuth 2.0 info
Setup the Token Expiry Date to be 1h after the access token was generated (or closer time to it in the :30 mins intervals it allows)
Initiated sync
Worked flawlessly until access token died (1h)
Received 401 error.

I tried both prod and sandbox, same error. I would appreciate your assistance a lot, thank you very much!

### Relevant log output

```shell
Configuration check failed
'Unable to connect to stream accounts - 401 Client Error: Unauthorized for url: https://oauth.platform.intuit.com/oauth2/v1/tokens/bearer'
```

### Contribute

- [ ] Yes, I want to contribute

**Comment by charles-astrafy** [comment]
Same issue here. It seems the mechanism to refresh the token is not working

**Comment by marcosmarxm** [comment]
@pabloescoder are you working in other issue related to Quickbooks? Can you take a look into this one?

**Comment by pabloescoder** [comment]
@marcosmarxm I've made changes for the other issue and raised a PR, waiting for that to get merged because after those changes the refresh logic is working locally for me, there's an issue with the regression tests which is a blocker. Once that's resolved and that PR is merged, this should ideally be solved. 
If it's not resolved even after that, then I'll definitely take a look into what is going wrong.

**Comment by tomsweeting** [comment]
Originally posted this [here](https://github.com/airbytehq/airbyte/issues/44471) but I think I posed to the wrong thread, so adding it here for visibility.

I have encountered this same issue using latest versions (Arbyte 1.3.0 + Quickbooks Connector 3.0.26)

The initial sync works fine, however after 1 hour once the initial token expires, the sync fails with error: `Failure in source: Unauthorized. Please ensure you are authenticated correctly.`

Steps taken to replicate the issue this morning:

- Generated new tokens ~ 10:30AM
- Ran three manual syncs within the hour, all succeeded
- Set up cron schedule
- On first schedule trigger at 12:01PM the connection no longer works

<img width="1231" alt="Image" src="https://github.com/user-attachments/assets/234cd99f-fd38-4ebe-970f-37178e0dfd92" />

**Comment by pabloescoder** [comment]
Hey @tomsweeting  
We just released a new version of QB (v4.0.0) with the fix. 
It's a major version upgrade because the credentials object was removed from the config and all the config fields were moved to the root level so you might need to repopulate the config fields. 
TLDR: QB v4.0.0 solves this issue!

**Comment by niteshsharoff** [comment]
I am (extremely) proud to say this synced on it's own for me today in Cloud! Quickbooks finally works automatically! 🥳

**Comment by tomsweeting** [comment]
Hey @pabloescoder thanks for the heads-up - that's exciting news! I will try upgrading to 4.0.0 tomorrow to test it out.

**Comment by chase-allenops** [comment]
@tomsweeting did you get it working? I'm still running an old version of the QB connector and it recently started failing.

**Comment by chase-allenops** [comment]
@tomsweeting thanks for the in-depth response. Sounds like I need to bite the bullet and write my own QBO extractor.

**Comment by kevinbretthauer** [comment]
I was experiencing the 24 hours that was fixed by 4.0. As of March 12th, I am experiencing a similar error again. I can get the system to sync if I manually refresh the authentication token in the QBO developer playground and then put it into airbye each day. 

Below is the log from the error I get if I do not manually replace the authentication token. Appreciate any advice.

2025-03-14 10:35:45 info 
2025-03-14 10:35:45 info ----- START CHECK -----
2025-03-14 10:35:45 info 
2025-03-14 10:35:50 info Connector exited, processing output
2025-03-14 10:35:50 info Output file jobOutput.json found
2025-03-14 10:35:50 info Connector exited with exit code 0
2025-03-14 10:35:50 info Reading messages from protocol version 0.2.0
2025-03-14 10:35:50 error Encountered an error trying to connect to stream accounts. Error: 
 Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/declarative/checks/check_stream.py", line 46, in check_connection
    stream_is_available, reason = availability_strategy.check_availability(
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/streams/http/availability_strategy.py", line 48, in check_availability
    self.get_first_record_for_slice(stream, stream_slice)
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/streams/availability_strategy.py", line 81, in get_first_record_for_slice
    return next(records_for_slice)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/declarative/declarative_stream.py", line 154, in read_records
    yield from self.retriever.read_records(self.get_json_schema(), stream_slice)  # type: ignore # records are of the correct type
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 457, in read_records
    for stream_data in self._read_pages(record_generator, self.state, _slice):
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 354, in _read_pages
    response = self._fetch_next_page(stream_state, stream_slice, next_page_token)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 310, in _fetch_next_page
    return self.requester.send_request(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/declarative/requesters/http_requester.py", line 411, in send_request
    headers=self._request_headers(
            ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/declarative/requesters/http_requester.py", line 257, in _request_headers
    headers = self._get_request_options(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/declarative/requesters/http_requester.py", line 240, in _get_request_options
    auth_options_method(),
    ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py", line 85, in get_auth_header
    token = self.access_token if self._is_access_token_flow else self.get_access_token()
                                                                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/streams/http/requests_native_auth/oauth.py", line 347, in get_access_token
    self.refresh_access_token()
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/streams/http/requests_native_auth/oauth.py", line 365, in refresh_access_token
    response_json = self._make_handled_request()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/backoff/_sync.py", line 105, in retry
    ret = target(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py", line 214, in _make_handled_request
    response.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://oauth.platform.intuit.com/oauth2/v1/tokens/bearer

2025-03-14 10:35:50 error Check failed
2025-03-14 10:35:50 info Checking for optional control message...
2025-03-14 10:35:50 info Writing output of cf9c4355-b171-4477-8f2d-6c5cc5fc8b7e_9243fea1-152c-4eb1-8c28-f7ba23483deb_0_check to the doc store
2025-03-14 10:35:50 info Marking workload cf9c4355-b171-4477-8f2d-6c5cc5fc8b7e_9243fea1-152c-4eb1-8c28-f7ba23483deb_0_check as successful
2025-03-14 10:35:51 info 
2025-03-14 10:35:51 info Deliberately exiting process with code 0.
2025-03-14 10:35:51 info ----- END CHECK -----
2025-03-14 10:35:51 info

**Comment by kevinbretthauer** [comment]
This issue still exists. I tried resyncing and the error still exists. This is a critical point of our data ware house. There is at least one other user who has commented and is still experiencing this issue. I would appreciate it if someone could look into it. Here is the lastest logs:

2025-11-03 13:34:39 replication-orchestrator INFO Failures: [ {
  "failureOrigin" : "source",
  "failureType" : "system_error",
  "internalMessage" : "ISO 8601 time designator 'T' missing. Unable to parse datetime string 'D'",
  "externalMessage" : "Something went wrong in the connector. See the logs for more details.",
  "metadata" : {
    "attemptNumber" : 4,
    "jobId" : 56122656,
    "from_trace_message" : true,
    "connector_command" : "read"
  },
  "stacktrace" : "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.13/site-packages/isodate/isodatetime.py\", line 51, in parse_datetime\n    datestring, timestring = datetimestring.split('T')\n    ^^^^^^^^^^^^^^^^^^^^^^\nValueError: not enough values to unpack (expected 2, got 1)\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/airbyte/integration_code/main.py\", line 4, in <module>\n    run()\n    ~~~^^\n  File \"/airbyte/integration_code/source_declarative_manifest/run.py\", line 307, in run\n    handle_command(args)\n    ~~~~~~~~~~~~~~^^^^^^\n  File \"/airbyte/integration_code/source_declarative_manifest/run.py\", line 91, in handle_command\n    handle_local_manifest_command(args)\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^\n  File \"/airbyte/integration_code/source_declarative_manifest/run.py\", line 129, in handle_local_manifest_command\n    launch(\n    ~~~~~~^\n        source=source,\n        ^^^^^^^^^^^^^^\n        args=args,\n        ^^^^^^^^^^\n    )\n    ^\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/entrypoint.py\", line 377, in launch\n    for message in source_entrypoint.run(parsed_args):\n                   ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/entrypoint.py\", line 207, in run\n    yield from map(\n    ...<2 lines>...\n    )\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/entrypoint.py\", line 280, in read\n    for message in self.source.read(self.logger, config, catalog, state):\n                   ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/concurrent_declarative_source.py\", line 374, in read\n    streams=self.streams(config=self._config),  # type: ignore  # We are migrating away from the DeclarativeStream implementation and streams() only returns the concurrent-compatible AbstractStream. To preserve compatibility, we retain the existing method interface\n            ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/concurrent_declarative_source.py\", line 409, in streams\n    self._constructor.create_component(\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        (\n        ^\n    ...<6 lines>...\n        emit_connector_builder_messages=self._emit_connector_builder_messages,\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py\", line 857, in create_component\n    return self._create_component_from_model(\n           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        model=declarative_component_model, config=config, **kwargs\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py\", line 874, in _create_component_from_model\n    return component_constructor(model=model, config=config, **kwargs)\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py\", line 1999, in create_default_stream\n    concurrent_cursor = self._build_concurrent_cursor(model, partition_router, config)\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py\", line 2249, in _build_concurrent_cursor\n    return self.create_concurrent_cursor_from_perpartition_cursor(  # type: ignore # This is a known issue that we are creating and returning a ConcurrentCursor which does not technically implement the (low-code) StreamSlicer. However, (low-code) StreamSlicer and ConcurrentCursor both implement StreamSlicer.stream_slices() which is the primary method needed for checkpointing\n           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n        state_manager=self._connector_state_manager,\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    ...<7 lines>...\n        attempt_to_create_cursor_if_not_provided=True,  # FIXME can we remove that now?\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py\", line 1675, in create_concurrent_cursor_from_perpartition_cursor\n    return ConcurrentPerPartitionCursor(\n        cursor_factory=cursor_factory,\n    ...<9 lines>...\n        attempt_to_create_cursor_if_not_provided=attempt_to_create_cursor_if_not_provided,\n    )\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/incremental/concurrent_partition_cursor.py\", line 187, in __init__\n    self._set_initial_state(stream_state)\n    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/incremental/concurrent_partition_cursor.py\", line 478, in _set_initial_state\n    self._create_cursor(state[\"cursor\"])\n    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/incremental/concurrent_partition_cursor.py\", line 580, in _create_cursor\n    cursor = self._cursor_factory.create(\n        stream_state=deepcopy(cursor_state),\n        runtime_lookback_window=timedelta(seconds=runtime_lookback_window),\n    )\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/incremental/concurrent_partition_cursor.py\", line 101, in create\n    return self._create_function(\n           ~~~~~~~~~~~~~~~~~~~~~^\n        stream_state=stream_state, runtime_lookback_window=runtime_lookback_window\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/usr/local/lib/python3.13/site-packages/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py\", line 1370, in create_concurrent_cursor_from_datetime_based_cursor\n    lookback_window = parse_duration(evaluated_lookback_window)\n  File \"/usr/local/lib/python3.13/site-packages/isodate/isoduration.py\", line 91, in parse_duration\n    durdt = parse_datetime(datestring[1:])\n  File \"/usr/local/lib/python3.13/site-packages/isodate/isodatetime.py\", line 53, in parse_datetime\n    raise ISO8601Error(\"ISO 8601 time designator 'T' missing. Unable to\"\n                       \" parse datetime string %r\" % datetimestring)\nisodate.isoerror.ISO8601Error: ISO 8601 time designator 'T' missing. Unable to parse datetime string 'D'\n",
  "timestamp" : 1762205661729
}, {
  "failureOrigin" : "destination",
  "failureType" : "transient_error",
  "internalMessage" : "io.airbyte.cdk.TransientErrorException: Input was fully read, but some streams did not receive a terminal stream status message. If the destination did not encounter other errors, this likely indicates an error in the source or platform. Streams without a status message: [fc_calendly.groups, fc_calendly.api_user, fc_calendly.event_types, fc_calendly.scheduled_events, fc_calendly.organization_memberships]",
  "externalMessage" : "Input was fully read, but some streams did not receive a terminal stream status message. If the destination did not encounter other errors, this likely indicates an error in the source or platform. Streams without a status message: [fc_calendly.groups, fc_calendly.api_user, fc_calendly.event_types, fc_calendly.scheduled_events, fc_calendly.organization_memberships]",
  "metadata" : {
    "attemptNumber" : 4,
    "jobId" : 56122656,
    "from_trace_message" : true,
    "connector_command" : "write"
  },
  "stacktrace" : "io.airbyte.cdk.TransientErrorException: Input was fully read, but some streams did not receive a terminal stream status message. If the destination did not encounter other errors, this likely indicates an error in the source or platform. Streams without a status message: [fc_calendly.groups, fc_calendly.api_user, fc_calendly.event_types, fc_calendly.scheduled_events, fc_calendly.organization_memberships]\n\tat io.airbyte.cdk.load.state.SyncManager.markInputConsumed(SyncManager.kt:116)\n\tat io.airbyte.cdk.load.state.PipelineEventBookkeepingRouter.close(PipelineEventBookkeepingRouter.kt:300)\n\tat io.airbyte.cdk.load.task.internal.InputConsumerTask.execute(InputConsumerTask.kt:117)\n\tat io.airbyte.cdk.load.task.DestinationTaskLauncher$WrappedTask.execute(DestinationTaskLauncher.kt:124)\n\tat io.airbyte.cdk.load.task.TaskScopeProvider$launch$job$1.invokeSuspend(TaskScopeProvider.kt:35)\n\tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)\n\tat kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)\n\tat kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)\n\tat kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)\n\tat kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)\n",
  "timestamp" : 1762205678726
}, {
  "failureOrigin" : "source",
  "internalMessage" : "Source process exited with non-zero exit code 1",
  "externalMessage" : "Something went wrong within the source connector",
  "metadata" : {
    "attemptNumber" : 4,
    "jobId" : 56122656,
    "connector_command" : "read"
  },
  "stacktrace" : "io.airbyte.workers.internal.exception.SourceException: Source process exited with non-zero exit code 1\n\tat io.airbyte.container.orchestrator.worker.SourceReader.run(ReplicationTask.kt:210)\n\tat io.airbyte.container.orchestrator.worker.ReplicationWorker$runJobs$2$tasks$1$1.invokeSuspend(ReplicationWorker.kt:152)\n\tat io.airbyte.container.orchestrator.worker.ReplicationWorker$runJobs$2$tasks$1$1.invoke(ReplicationWorker.kt)\n\tat io.airbyte.container.orchestrator.worker.ReplicationWorker$runJobs$2$tasks$1$1.invoke(ReplicationWorker.kt)\n\tat io.airbyte.container.orchestrator.worker.util.AsyncUtils$runAsync$1$1.invokeSuspend(AsyncUtils.kt:22)\n\tat io.airbyte.container.orchestrator.worker.util.AsyncUtils$runAsync$1$1.invoke(AsyncUtils.kt)\n\tat io.airbyte.container.orchestrator.worker.util.AsyncUtils$runAsync$1$1.invoke(AsyncUtils.kt)\n\tat kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)\n\tat kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:43)\n\tat kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:165)\n\tat kotlinx.coroutines.BuildersKt.withContext(Unknown Source)\n\tat io.airbyte.container.orchestrator.worker.util.AsyncUtils$runAsync$1.invokeSuspend(AsyncUtils.kt:21)\n\tat kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)\n\tat kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)\n\tat io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:141)\n\tat io.micrometer.core.instrument.Timer.lambda$wrap$2(Timer.java:199)\n\tat datadog.trace.bootstrap.instrumentation.java.concurrent.Wrapper.run(Wrapper.java:47)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\n",
  "timestamp" : 1762205679768
} ]
2025-11-03 13:34:39 replication-orchestrator INFO 
2025-11-03 13:34:39 replication-orchestrator INFO Closing io.micronaut.configuration.metrics.binder.executor.ExecutorServiceMetricsBinder$2@3c0a1c1c
2025-11-03 13:34:39 replication-orchestrator INFO ----- END REPLICATION -----
2025-11-03 13:34:39 replication-orchestrator INFO 
2025-11-03 13:34:39 replication-orchestrator INFO Closing io.micronaut.configuration.metrics.binder.executor.ExecutorServiceMetricsBinder$2@6d5a43c1
2025-11-03 13:34:39 replication-orchestrator INFO Closing io.airbyte.container.orchestrator.worker.io.LocalContainerAirbyteDestination@1c07365c
2025-11-03 13:34:40 replication-orchestrator ERROR Error closing resource io.airbyte.container.orchestrator.worker.io.LocalContainerAirbyteDestination@1c07365c; recording failure but continuing.
2025-11-03 13:34:40 replication-orchestrator INFO Closing io.airbyte.container.orchestrator.worker.io.LocalContainerAirbyteSource@55e60c72
2025-11-03 13:34:40 replication-orchestrator ERROR Error closing resource io.airbyte.container.orchestrator.worker.io.LocalContainerAirbyteSource@55e60c72; recording failure but continuing.
2025-11-03 13:34:40 replication-orchestrator INFO Closing io.airbyte.container.orchestrator.worker.RecordSchemaValidator@7c909fea
2025-11-03 13:34:40 replication-orchestrator INFO Closing io.airbyte.container.orchestrator.persistence.SyncPersistenceImpl@40e5736f
2025-11-03 13:34:42 replication-orchestrator INFO Returning output...
2025-11-03 13:34:42 platform INFO 
----- START POST REPLICATION OPERATIONS -----

2025-11-03 13:34:42 platform INFO No post-replication operation(s) to perform.
2025-11-03 13:34:42 platform INFO 
----- END POST REPLICATION OPERATIONS -----

2025-11-03 13:34:43 platform INFO Retry State: RetryManager(completeFailureBackoffPolicy=BackoffPolicy(minInterval=PT10S, maxInterval=PT30M, base=3), partialFailureBackoffPolicy=null, successiveCompleteFailureLimit=5, successivePartialFailureLimit=1000, totalCompleteFailureLimit=5, totalPartialFailureLimit=20, successiveCompleteFailures=5, successivePartialFailures=0, totalCompleteFailures=5, totalPartialFailures=0)
 Backoff before next attempt: 13 minutes 30 seconds
2025-11-03 13:34:43 platform ERROR Failing job: 56122656, reason: Job failed after too many retries for connection 180265e2-28c8-4693-b5a6-050ada25b82a
